syntax = "proto3";

package Proyecto1Datos2;

// Definición del servicio GRPC
service MemoryManager {
  rpc Create(CreateRequest) returns (CreateResponse);
  rpc Set(SetRequest) returns (SetResponse);
  rpc Get(GetRequest) returns (GetResponse);
  rpc IncreaseRefCount(RefCountRequest) returns (RefCountResponse);
  rpc DecreaseRefCount(RefCountRequest) returns (RefCountResponse);
}

// Mensajes de solicitud y respuesta

message CreateRequest {
  int32 id = 1;
  int32 size = 2;  // ✔ Tamaño de la memoria en bytes

  //string type = 3;
}

message CreateResponse {
  int32 id = 1;

}

message SetRequest {
  int32 id = 1;
  string value = 2;
}

message SetResponse {
  bool success = 1;
}

message GetRequest {
  int32 id = 1;
}

message GetResponse {
  string value = 1;
}

message RefCountRequest {
  int32 id = 1;
}

message RefCountResponse {
  bool success = 1;
}
